src/webnn/native/dml/DMLUtils.h

```diff
@@ -144,32 +200,86 @@ namespace webnn::native::dml {
                 D3D12_TEXTURE_LAYOUT_ROW_MAJOR,  flags};
     };
 
-    template <typename T>
-    T RoundUpToMultiple(T value, T multiple) {
-        static_assert(std::is_integral_v<T>);
-
-        T remainder = value % multiple;
-        if (remainder != 0) {
-            value += multiple - remainder;
-        }
+    struct CompiledGraph {
+        CompiledGraph(const ComPtr<IDMLDevice>& device,
+                      const DML_GRAPH_DESC& graphDesc,
+                      const std::vector<std::shared_ptr<InputNode>>& inputs,
+                      const std::vector<Node>& outputs,
+                      DML_EXECUTION_FLAGS flag = DML_EXECUTION_FLAG_NONE)
+            : inputNodes(inputs), outputNodes(outputs) {
+            ComPtr<IDMLDevice1> device1;
+            WEBNN_CHECK(device.Get()->QueryInterface(IID_PPV_ARGS(&device1)));
```

Comment:
It's a bad idea to call non-trivial functions which can fail within a constructor. Should they error, this leaves the object in a corrupted state which causes undefined program behavior. 

I suggest to use `HRESULT CreateCompileGraph(..)` which *always* creates a valid `new CompiledGraph` once those non-trivial functions succeed.

Reply:
OK, I will!